### PR TITLE
Added the no-unused-vars argsIgnorePattern

### PR DESCRIPTION
**Pull request by Eruant**
Enables use of underscore to show unused variables.

```
const myFunc = _ => {
  // ...
}

const myOtherFunc = (_a, b) => {
  // ...
}
```

[](https://api.gh-polls.com/poll/01CCGJHF3RP5DA61TYJE29DC2F/Yes/vote)
[](https://api.gh-polls.com/poll/01CCGJHF3RP5DA61TYJE29DC2F/No/vote)

### CODE REVIEW

**Comment by simontabor** [comment]
@Eruant is it be possible to enforce never using vars prefixed with `_`?

**Comment by Eruant** [comment]
@simontabor My understanding is that all variables declared should be used. If you want to declare a variable that will be ignored you can use a `_` to get around the rule.

Does that answer your question, or did you mean adding in separate rule? ([like this rule](https://eslint.org/docs/rules/no-underscore-dangle))

**Comment by simontabor** [comment]
@Eruant Sorry, no I agree I'm just wondering if we can have it so eslint will highlight this incorrect usage
```js
const myOtherFunc = (_a, b) => {
  // ...

  myFunc(_a) // not allowed to actually use _a
}
```

**Comment by Eruant** [comment]
@simontabor  Like changing ignore rule from `^_` to `^_$`?

**Comment by Eruant** [comment]
I think I've just got what you are saying, and I don't know if any rule that will enforce it....

**Comment by antony** [comment]
I feel like an unused variable placeholder is a code smell.

File: node.js [hunk 1]
```diff
@@ -84,6 +84,7 @@ module.exports = {
     'space-infix-ops': error,
     'prefer-const': warn,
     'prefer-rest-params': error,
-    'prefer-spread': error
+    'prefer-spread': error,
+    'no-unused-vars': [ error, { argsIgnorePattern: '^_' } ]
```
Comment (alixeb) [inline node.js]: it should use => https://eslint.org/docs/rules/no-unused-vars#args
`{ "args": "after-used" }`

Comment (Eruant) [reply node.js]: That would result in a different behaviour as far as I can tell.

**Comment by Eruant** [comment]
Closing this, as it's seems most people are against it.

**Pull request closed without merging**